cmd/serv.go

```diff
@@ -72,6 +73,7 @@ var (
 		"git-receive-pack":   models.AccessModeWrite,
 		lfsAuthenticateVerb:  models.AccessModeNone,
 	}
+	alphaDashDotPattern = regexp.MustCompile(`[^\w-\.]`)
```

Comment:
It feel that this should belong somewhere in models package or in modules/validation/.
But that a more global problem that input validation are scattered across gitea code.

Reply:
I agree, when writing this I felt like it would be nice to have a central validation method, but when I looked this validation is actually in Macaron code and we use a binding for it.  
I think there's one other place that uses this pattern, but I forget what for off the top of my head.